test/core/end2end/fake_resolver.h

```diff
@@ -32,8 +32,23 @@
 #ifndef GRPC_TEST_CORE_END2END_FAKE_RESOLVER_H
 #define GRPC_TEST_CORE_END2END_FAKE_RESOLVER_H
 
+#include "src/core/lib/channel/channel_args.h"
 #include "test/core/util/test_config.h"
 
 void grpc_fake_resolver_init();
 
+// Return a \a grpc_arg enabling LB.
+grpc_arg grpc_fake_resolver_lb_enabled_arg();
```

Comment:
Yes. The intention was to do that in a separate PR (the one for LB policies server updates), but you have a point that that mechanism can encompass everything. I'll re-purpose this PR to use the general approach for everything.